usecases/seed_usecase.go

```diff
@@ -21,86 +25,84 @@ type SeedUseCase struct {
+	return nil
+}
 
-	// assign test s3 bucket name to zorg organization
-	testBucketName := "marble-backend-export-scheduled-execution-test"
-	err = usecase.organizationRepository.UpdateOrganization(ctx, exec, models.UpdateOrganizationInput{
-		Id:                         zorgOrganizationId,
-		ExportScheduledExecutionS3: &testBucketName,
-	})
-	if err != nil {
-		return err
+// This method is supposed to be used as a script when starting the server, not from the API
+// Hence it does not enforce any authorization, since there is also no user credentials context
+func (usecase *SeedUseCase) CreateOrgAndUser(ctx context.Context, input models.InitOrgInput) error {
+	if input.OrgName == "" {
+		return errors.New("Cannot create organization or org admin with empty name in CreateOrgAndUser")
 	}
-
-	// add Admin user Jean-Baptiste Emanuel Zorg
-	_, err = usecase.userRepository.CreateUser(ctx, exec, models.CreateUser{
-		Email:          "jbe@zorg.com", // Jean-Baptiste Emanuel Zorg
-		Role:           models.ADMIN,
-		OrganizationId: zorgOrganizationId,
-		FirstName:      "Jean-Baptiste Emanuel",
-		LastName:       "Zorg",
-	})
-	if repositories.IsUniqueViolationError(err) {
-		err = nil
+	if input.AdminEmail != "" {
+		_, err := mail.ParseAddress(input.AdminEmail)
+		if err != nil {
+			return errors.New(fmt.Sprintf("Invalid email address %s in CreateOrgAndUser", input.AdminEmail))
+		}
 	}
+	logger := utils.LoggerFromContext(ctx)
+	exec := usecase.executorFactory.NewExecutor()
+
+	var targetOrg models.Organization
+	allOrgs, err := usecase.organizationRepository.AllOrganizations(ctx, exec)
 	if err != nil {
 		return err
 	}
-
-	newCustomListId := "d6643d7e-c973-4899-a9a8-805f868ef90a"
-
-	err = usecase.customListRepository.CreateCustomList(ctx, exec, models.CreateCustomListInput{
-		Name:        "zorg custom list",
-		Description: "Need a whitelist or blacklist ? The list is your friend :)",
-	}, zorgOrganizationId, newCustomListId)
-
-	if err == nil {
-		// add some values to the hardcoded custom list
-		addCustomListValueInput := models.AddCustomListValueInput{
-			CustomListId: newCustomListId,
-			Value:        "Welcome",
+	for _, org := range allOrgs {
+		if org.Name == input.OrgName {
+			targetOrg = org
+			logger.InfoContext(
+				ctx,
+				fmt.Sprintf("Organization %s already exists for name %s", targetOrg.Id, input.OrgName),
+			)
+			break
 		}
-		usecase.customListRepository.AddCustomListValue(ctx, exec, addCustomListValueInput, uuid.NewString())
-		addCustomListValueInput.Value = "to"
-		usecase.customListRepository.AddCustomListValue(ctx, exec, addCustomListValueInput, uuid.NewString())
-		addCustomListValueInput.Value = "marble"
-		usecase.customListRepository.AddCustomListValue(ctx, exec, addCustomListValueInput, uuid.NewString())
 	}
 
-	if repositories.IsUniqueViolationError(err) {
-		err = nil
+	if targetOrg.Id == "" {
+		organizationId := uuid.NewString()
+		targetOrg, err = usecase.organizationCreator.CreateOrganizationWithId(
```

Comment:
Honnestly, since we no longuer need to use a fixed id in order to find if the seed org already exists, I would drop this `CreateOrganizationWithId` method.

Finally, we can directly merge this inside the only part of the code using this: the org repository

Reply:
well yea fair point 👍